### PR TITLE
feat(components): add ErrorBoundary across core + Leptos + Dioxus

### LINKED ISSUE #197: task: Implement ArsErrorBoundary component in ars-dioxus
**Foundation** | **Points: 2** | **Epic: #9 (Dioxus adapter)**

## Goal

Implement the `ArsErrorBoundary` wrapper component that uses Dioxus's `ErrorBoundary` to gracefully handle machine panics or unexpected state transitions, rendering an accessible error fallback UI.

## Layer

Adapter

## Framework

Dioxus

## Test tier

Unit

## Depends on

None (only uses Dioxus `ErrorBoundary` primitive)

## Spec refs

- `spec/foundation/09-adapter-dioxus.md` §21 "Error Boundary Pattern" (L2714)

## Files to create/modify

- `crates/ars-dioxus/src/error_boundary.rs` (new) — `ArsErrorBoundary` component
- `crates/ars-dioxus/src/lib.rs` — wire new module, update exports
- `crates/ars-dioxus/src/prelude.rs` — re-export `ArsErrorBoundary` for end users

## Tests to add first

- Unit test for `ArsErrorBoundary` rendering children when no error occurs.
- Unit test for `ArsErrorBoundary` rendering fallback UI with `data-ars-error="true"` and `role="alert"` when a child component throws.
- Unit test for error message display via `ErrorContext::error()` (returns `Option`).
- Unit test verifying the fallback does not propagate the error further (boundary catches it).

## Acceptance criteria

- `ArsErrorBoundary` component accepting `children: Element` prop.
- Wraps children in Dioxus `ErrorBoundary` with `handle_error` callback.
- Error fallback renders a `div` with:
  - `data-ars-error="true"` attribute for test targeting.
  - `role="alert"` for accessibility (screen readers announce errors).
  - A paragraph describing the error: "A component encountered an error."
  - A second paragraph with the error message from `ErrorContext::error()` (using `.map(|e| format!("{e}")).unwrap_or_default()`).
- Exported in `ars_dioxus::prelude` for end-user consumption.
- Zero warnings under workspace lint config.
- All public items have `///` doc comments.

## Spec impact

No spec change required.

### LINKED ISSUE #629: task: Implement ArsErrorBoundary component in ars-leptos
**Foundation** | **Points: 2** | **Epic: Leptos adapter (parity with #197)**

## Goal

Implement the `ArsErrorBoundary` wrapper that uses Leptos `ErrorBoundary` to gracefully handle machine panics, mirroring the Dioxus implementation tracked in #197 and required by the prelude-symmetry rule in CLAUDE.md.

## Layer

Adapter

## Framework

Leptos

## Test tier

Unit

## Depends on

#197 (sibling — must land in the same PR per the prelude-symmetry rule).

## Spec refs

- `spec/foundation/08-adapter-leptos.md` §17 "Error Boundary Pattern"

## Files to create/modify

- `crates/ars-leptos/src/error_boundary.rs` (new)
- `crates/ars-leptos/src/lib.rs`
- `crates/ars-leptos/src/prelude.rs`

## Tests to add first

- Children render when no error.
- Fallback renders `data-ars-error="true"`, `role="alert"`, the static message, and a `<ul>` of error strings when a child returns `Err`.
- Boundary catches the error (does not propagate past the wrapper).

## Acceptance criteria

- `ArsErrorBoundary` accepting `children: Children`.
- Wraps Leptos `ErrorBoundary` with the §17 fallback closure.
- Exposed via `ars_leptos::prelude`.
- Zero warnings; all public items documented.

## Spec impact

None.

### PR DESCRIPTION
**Pull request by fogodev**
## Summary

- Implements `error_boundary::Boundary` with adapter-symmetric accessible fallback (`role="alert"`, localized heading, `<ul>`/`<li>` error list with `data-ars-error-count`) backed by a framework-agnostic `Messages` / `Part` / `Api` core in `ars-components`.
- Both adapters expose a `default_fallback` helper for consumers who use the framework-native `<ErrorBoundary>` directly, plus optional props for fallback override, telemetry (`on_error`), and message-bundle override resolved through `ArsProvider`'s i18n registry.
- Foundation §17 (Leptos) / §21 (Dioxus) replaced with thin shim sections cross-referencing the new canonical spec at `spec/components/utility/error-boundary.md` and per-adapter specs at `spec/{leptos,dioxus}-components/utility/`.
- New `cargo xtask spec compile-snippets [--fix]` tool parses every Rust code block in the spec corpus through `syn::parse_file`; integrated into `cargo xci` as a new step. Re-tagged 534 pre-existing partial spec snippets with `rust,no_check`.
- `CLAUDE.md` / `AGENTS.md` coverage section corrected to reflect the merged native + wasm coverage pipeline (the previous "can't produce lcov" wording was outdated).

## Test plan

- [x] `cargo xci fmt check clippy unit integration adapter adapter-parity snapshot-count spec-compile-snippets error-variant-coverage` — all 10 native steps pass
- [x] `cargo xci feature-matrix-leptos feature-matrix-dioxus release mutual-exclusion` — all 4 steps pass
- [x] `cargo test -p ars-components --doc utility::error_boundary` — 2 doc tests pass
- [x] `cargo test -p ars-components --test proptest_state_machines proptest_error_boundary -- --include-ignored` — 4 proptests pass
- [x] 97 tests total: 14 framework-agnostic core + 5 proptests + 24 Dioxus SSR + 3 Dioxus desktop + 24 Leptos SSR + 3 Leptos inline + 22 xtask `compile_snippets` + 1 xtask spec-corpus integration + 2 doc tests
- [x] Coverage: 100% on the framework-agnostic `error_boundary.rs`, 98.7% lines on `xtask/spec/compile_snippets.rs`
- [x] All snapshot files reviewed (12 new `.snap` files: 6 framework-agnostic AttrMap + 2 Dioxus HTML + 2 Leptos HTML + 2 unchanged)

## Notable design points

- **Adapter symmetry** — Dioxus `ErrorContext` exposes a single `Option<CapturedError>`; Leptos `Errors` is a multi-error map. The wrapper normalizes both into the canonical `<ul>`/`<li>` shape so consumer DOM/CSS/test selectors are framework-agnostic.
- **`Messages` priority** — direct prop > `ArsProvider` i18n registry (locale-aware) > `Messages::default()` (English). All three branches are unit-tested.
- **WAI-ARIA contract** — `role="alert"` + explicit `aria-live="assertive"` + `aria-atomic="true"`. Nested-Boundary case: inner catches, outer stays idle (single alert region).
- **XSS safe** — error `Display` text is HTML-escaped by both frameworks; tests assert the contract using `<script>alert('xss')</script>` payload.
- **Recovery** — Dioxus `ErrorContext::clear_errors()` and Leptos `errors.update(|e| *e = Errors::default())` exposed through custom fallbacks; recovery test included for Dioxus end-to-end.

Closes #197
Closes #629

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/ars-leptos/src/error_boundary.rs [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+fn run_fallback(
+    errors: ArcRwSignal<Errors>,
+    on_error: Option<Callback<CapturedError>>,
+    fallback: Option<FallbackHandler>,
+    heading: String,
+) -> AnyView {
+    if let Some(handler) = on_error {
+        for (_, error) in errors.get() {
+            handler.run(error.clone());
+        }
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-leptos/src/error_boundary.rs:156]: Fixed in 41e9d70. `Boundary` now carries a seen `ErrorId` set across fallback re-renders, so `on_error` only fires for newly observed errors in the episode instead of replaying unchanged entries. I added `run_fallback_deduplicates_telemetry_across_rerenders` for the regression and verified it with `cargo test -p ars-leptos --features ssr error_boundary`, focused `cargo llvm-cov test -p ars-leptos --features ssr --summary-only error_boundary`, and full `cargo xci` using the pinned local browser/wasm runner.

File: crates/ars-dioxus/src/error_boundary.rs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+    let resolved_messages = use_messages(messages.as_ref(), Some(&resolved_locale));
+
+    let heading = (resolved_messages.message)(&resolved_locale);
+
+    rsx! {
+        ErrorBoundary {
+            handle_error: move |ctx: ErrorContext| {
+                if let (Some(error), Some(handler)) = (ctx.error(), on_error.as_ref()) {
+                    handler.call(error);
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-dioxus/src/error_boundary.rs]: Fixed in 64e2783. Dioxus `Boundary` now keeps a per-boundary seen set keyed by the captured error identity, so fallback re-renders do not replay `on_error` for the same episode. Added `should_emit_new_error_deduplicates_same_captured_error` to pin same-error dedupe while still allowing a distinct captured error with the same display text to emit. Verified with `cargo test -p ars-dioxus --features desktop error_boundary`, focused `cargo llvm-cov test -p ars-dioxus --features desktop --summary-only error_boundary`, and full local `cargo xci`.

File: xtask/src/spec/compile_snippets.rs [hunk 3]
```diff
@@ -0,0 +1,891 @@
...
+        }
+
+        if in_block {
+            current.push_str(line);
+
+            current.push('\n');
+        }
+    }
+
+    blocks
```
[the remark this answers is not in the document]
Comment (fogodev) [reply xtask/src/spec/compile_snippets.rs:340]: Fixed in 64e2783. `extract_rust_blocks` now flushes an open Rust fence at EOF instead of dropping it, so an unterminated final snippet is still parsed and reported. Added `flushes_unterminated_final_rust_block` for the regression. Verified with `cargo test -p xtask spec::compile_snippets`, focused `cargo llvm-cov test -p xtask --summary-only spec::compile_snippets`, and full local `cargo xci`.

File: crates/ars-dioxus/src/error_boundary.rs [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            },
+            {children}
+        }
+    }
+}
+
+fn should_emit_new_error(error: &CapturedError, seen_error_ptrs: &mut HashSet<usize>) -> bool {
+    let error_ptr = std::sync::Arc::as_ptr(&error.0) as usize;
+
+    seen_error_ptrs.insert(error_ptr)
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-dioxus/src/error_boundary.rs]: Fixed in 930144d. Dioxus now keeps a bounded latest-error identity instead of an append-only pointer set, so fallback rerenders still dedupe the same captured error while later episodes replace the cached identity. I tightened `should_emit_new_error_deduplicates_same_captured_error` to prove the cached identity is updated, and verified with `cargo test -p ars-dioxus --features desktop error_boundary`, focused Dioxus llvm-cov, `cargo insta test --unreferenced=reject`, and full local `cargo xci`.

File: crates/ars-leptos/src/error_boundary.rs [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+    seen_error_ids: &Mutex<HashSet<ErrorId>>,
+) -> AnyView {
+    if let Some(handler) = on_error {
+        let snapshot = errors.get().into_iter().collect::<Vec<_>>();
+
+        let mut seen = seen_error_ids.lock().expect("lock seen error ids");
+
+        for (id, error) in snapshot {
+            if seen.insert(id) {
+                handler.run(error.clone());
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-leptos/src/error_boundary.rs:154]: Fixed in 930144d. Leptos now prunes `seen_error_ids` to the IDs present in the current `Errors` snapshot before emitting telemetry, so a cleared/recovered snapshot removes stale IDs and a reused ID can emit for a new episode. Added `run_fallback_prunes_seen_ids_when_errors_clear` and verified with `cargo test -p ars-leptos --features ssr error_boundary`, focused Leptos llvm-cov, `cargo insta test --unreferenced=reject`, and full local `cargo xci`.

File: crates/ars-leptos/src/error_boundary.rs [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+            .map(|(id, _)| id.clone())
+            .collect::<HashSet<_>>();
+
+        let mut seen = seen_error_ids.lock().expect("lock seen error ids");
+
+        seen.retain(|id| current_ids.contains(id));
+
+        for (id, error) in snapshot {
+            if seen.insert(id) {
+                handler.run(error.clone());
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-leptos/src/error_boundary.rs:154]: Fixed in 33989e8. Leptos now tracks the last emitted error fingerprint per `ErrorId`, so a same-id error replacement emits telemetry while unchanged rerenders still dedupe and cleared snapshots still prune stale entries. Added `run_fallback_emits_when_same_id_receives_new_error_value`; verified with `cargo test -p ars-leptos --features ssr error_boundary`, focused llvm-cov, and full local `cargo xci`.

File: xtask/src/spec/compile_snippets.rs [hunk 7]
```diff
@@ -0,0 +1,914 @@
...
+        }
+
+        let new_info = if info.contains(',') {
+            // `rust,foo` → `rust,foo,no_check`
+            format!("{info},no_check")
+        } else {
+            format!("{info},no_check")
+        };
+
+        *line = format!("{leading}```{new_info}");
```
[the remark this answers is not in the document]
Comment (fogodev) [reply xtask/src/spec/compile_snippets.rs]: Fixed in 33989e8. The compile-snippets fixer now preserves the opener backtick run length when appending `no_check`, so wider fences remain wider after `--fix`. Added `rewrite_fences_to_no_check_preserves_wide_fence_width`; verified with `cargo test -p xtask spec::compile_snippets`, focused llvm-cov, and full local `cargo xci`.
